### PR TITLE
[build] make sure flatten-maven-plugin runs after maven-shade-plugin

### LINKED ISSUE #2600: [Improve][build] The connector module correctly execute `shaded` and `flatten`
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

After we have done `shaded`, we did not remove the shaded dependency when we `flattened` the pom, which makes the connector's jar is uber, but the connector's pom does not exclude the shaded dependencies; 
this will cause problems such as dependency conflicts.

image

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ashulin**
## Purpose of this pull request
make sure flatten-maven-plugin runs after maven-shade-plugin

issue: https://github.com/apache/incubator-seatunnel/issues/2600

The maven install command can be executed in parallel: 

> mvn clean install  -T 1C

image

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by EricJoy2048 — DISMISSED** [review]
LGTM

**Comment by hailin0** [comment]
LGTM

File: tools/dependencies/checkLicense.sh [hunk 1]
```diff
@@ -42,4 +40,4 @@ echo '=== Third party dependencies: ' && grep -vf self-modules.txt all-dependenc
 # command in target OS is different from what we used to sort the file `known-dependencies.txt`, i.e. "sort the two file
 # using the same command (and default arguments)"
 
-diff -w -B -U0 <(sort < tools/dependencies/known-dependencies.txt) <(sort < third-party-dependencies.txt)
+echo '=== diff dependencies: ' && diff -w -B -U0 <(sort < tools/dependencies/known-dependencies.txt) <(sort < third-party-dependencies.txt)
```
Comment (CalvinKirs) [inline tools/dependencies/checkLicense.sh]: hi, I disabled this in #2613 so we won't be going to execute this script

**Comment by TyrantLucifer** [comment]
LGTM. BTW,pls unified with pr #2613

File: tools/dependencies/checkLicense.sh (hunk 1 above)
Comment (ashulin) [reply tools/dependencies/checkLicense.sh]: @CalvinKirs PTAL.

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**